File: src/main/java/com/lessmarkup/interfaces/data/DomainModel.java
package com.lessmarkup.interfaces.data;

public interface DomainModel extends AutoCloseable {
    QueryBuilder query();
    void completeTransaction();
    <T extends DataObject> boolean update(T dataObject);
    <T extends DataObject> boolean create(T dataObject);
    <T extends DataObject> boolean delete(Class<T> type, long id);
    @Override
    void close();
}

File: src/main/scala/com/lessmarkup/interfaces/data/DomainModel.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.data

trait DomainModel extends AutoCloseable {
  def query: QueryBuilder
  def completeTransaction()
  def update[T <: DataObject](dataObject: T): Boolean
  def create[T <: DataObject](dataObject: T): Boolean
  def delete[T <: DataObject](dataType: Class[T], id: Long): Boolean
  def close()
}